packages/core/src/components/DismissableBanner/DismissableBanner.tsx

```diff
@@ -60,17 +63,39 @@ type Props = {
 
 const DismissableBanner: FC<Props> = ({ variant, /* setting, */ message }) => {
   // const [show, setShown, loading] = useSetting(setting);
-  const [show, setShown] = useState(true);
   const classes = useStyles();
+  const storageApi = useApi(storageApiRef);
+  const settingsStore = storageApi.forBucket('settings');
+  const rawDismissedBanners =
+    settingsStore.get<string[]>('dismissedBanners') ?? [];
+
+  const [dismissedBanners, setDismissedBanners] = useState(
+    new Set(rawDismissedBanners),
+  );
+
+  const observedItems = useObservable(
+    settingsStore.observe$<string[]>('dismissedBanners'),
+  );
+
+  useEffect(() => {
+    // Only want to run on updates, not first time
+    if (observedItems?.newValue) {
+      const currentValue = observedItems?.newValue ?? [];
+      setDismissedBanners(new Set(currentValue));
+    }
+  }, [observedItems?.newValue]);
 
   const handleClick = () => {
-    setShown(false);
+    settingsStore.set('dismissedBanners', [
+      ...dismissedBanners,
+      ReactDOMServer.renderToString(message),
```

Comment:
Wonder if we can use an `id` on the message instead of using the text content? Might give us more flexibility if we change the wording of an message and don’t want it to pop up again?

Reply:
Yeah that sounds like a much cleaner way of handling it.

Is there a particular id source you had in mind, or should I just add something like an id prop to the DismissableBanner component?

Reply:
`id` prop sounds perfect I think. Supports backwards compatibility too which at this stage might not be so important but good to keep in mind!

Reply:
+1 for `id` prop.